client/src/components/cartCol.jsx

```diff
@@ -68,45 +67,48 @@ class CartCol extends React.Component {
   }
 
   componentDidMount() {
+    // console.log('CDM react:', this.state)
     const { itemData, itemId } = this.state;
     this.getURL();
     //  this.getData(itemId);
   }
 
   getURL() {
-    const currentURL = window.location;
-    const itemId = currentURL.pathname.split('/')[2];
-    this.setState({ itemId, isLoading: false },
-      () => this.getData(itemId),
-      );
+    let queryUrl = window.location.search;
+    let urlParams = new URLSearchParams(queryUrl);
+    let itemId = urlParams.get('itemId');
```

Comment:
This could be a problem when we put all the services together on the proxy. The former teams strategy was to take the pathname, cut it up, and use the number value at the end of '/items/[number]' and what you are doing is using a url parameter that will look something like '/?itemId=[number]/ and by deciding to go this route (if the whole team doesn't refactor) we will run into issues.

Reply:
Good callout. I left the legacy code in there to use "/items/[number]" but had trouble accessing the front end without using url params. This is a "refactor required" piece of my code to bring it back to the original, but will definitely want to review in standup tonight.